src/main/java/wolox/training/controllers/UsersController.java

```diff
@@ -0,0 +1,79 @@
+package wolox.training.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UsersNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.Users;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UsersRepository;
+
+public class UsersController {
+
+	@Autowired
+	private UsersRepository usersRepository;
+
+	@Autowired
+	private BookRepository bookRepository;
+
+	@GetMapping
+	public String index() {
+		return "index";
+	}
+
+	@GetMapping("/list")
+	public Iterable<Users> findAll() {
+		return usersRepository.findAll();
+	}
+
+	@GetMapping("/name/{usersName}")
+	public Iterable<Users> findAllByName(@PathVariable String name) {
+		return usersRepository.findAllByName(name);
+	}
+
+	@GetMapping("/{id}")
+	public Users findById(@PathVariable Long id) {
+		return usersRepository.findById(id).orElseThrow(() -> new UsersNotFoundException("User Not Found", new Exception()));
+	}
+
+	@PostMapping
+	@ResponseStatus(HttpStatus.CREATED)
+	public Users create(@RequestBody Users user) {
+		return usersRepository.save(user);
+	}
+
+	@DeleteMapping("/{id}")
+	public void delete(@PathVariable Long id) throws Throwable {
+		Users user = usersRepository.findById(id).orElseThrow(() -> new UsersNotFoundException("User Not Found", new Exception()));
+		usersRepository.delete(user);
+	}
+
+	@PutMapping("/{id}")
+	public Users update(@RequestBody Users user, @PathVariable Long id) {
+		if (user.getId() != id) {
+			throw new UsersNotFoundException("User id not found", new Exception());
+		}
+		return usersRepository.save(user);
+	}
+	@PutMapping("/{id}")
+	protected void addBooks(@PathVariable Long userId, @PathVariable Long bookId) {
+		Book book = bookRepository.findById(bookId).orElseThrow(() -> new BookNotFoundException("Book Not Found", new Exception()));
+		Users user = usersRepository.findById(userId).orElseThrow(() -> new UsersNotFoundException("User Not Found", new Exception()));
+		user.setBooks(book);
+	}
+
+	@DeleteMapping("/{id}")
```

Comment:
Be careful as this endpoint is duplicated, the endpoint for deleting a book from the user's collection should be something like `@DeleteMapping("/{id}/{bookId}")` 😅